#
# Ubuntu: https://bugs.launchpad.net/ubuntu/+source/gnome-terminal/+bug/331462
# Upstream: http://bugzilla.gnome.org/show_bug.cgi?id=572549
# Patch: http://svn.gnome.org/viewvc/gnome-terminal/trunk/src/terminal-screen.c?view=patch&r1=3329&r2=3328&pathrev=3329
# Description: fixes memory corruption in gnome-terminal
#

diff -Nur -x '*.orig' -x '*~' gnome-terminal-2.25.91/src/terminal-screen.c gnome-terminal-2.25.91.new/src/terminal-screen.c
--- gnome-terminal-2.25.91/src/terminal-screen.c	2009-02-24 17:21:10.000000000 +0100
+++ gnome-terminal-2.25.91.new/src/terminal-screen.c	2009-02-24 17:22:11.000000000 +0100
@@ -1746,12 +1746,12 @@
   g_return_val_if_fail (TERMINAL_IS_SCREEN (screen), NULL);
 
   if (priv->pty_fd == -1)
-    return priv->initial_working_directory;
+    return g_strdup (priv->initial_working_directory);
 
   /* Get the foreground process ID */
   fgpid = tcgetpgrp (priv->pty_fd);
   if (fgpid == -1)
-    return priv->initial_working_directory;
+    return g_strdup (priv->initial_working_directory);
 
   /* Try to get the working directory using various OS-specific mechanisms */
   for (i = 0; i < G_N_ELEMENTS (patterns); ++i)
@@ -1791,7 +1791,7 @@
         }
     }
 
-  return priv->initial_working_directory;
+  return g_strdup (priv->initial_working_directory);
 }
 
 void
